flink-end-to-end-tests/flink-bucketing-sink-test/src/main/java/org/apache/flink/streaming/tests/BucketingSinkTestProgram.java

```diff
@@ -71,7 +71,11 @@ public static void main(String[] args) throws Exception {
 
 		// define bucketing sink to emit the result
 		BucketingSink<Tuple4<Integer, Long, Integer, String>> sink = new BucketingSink<Tuple4<Integer, Long, Integer, String>>(outputPath)
-			.setBucketer(new KeyBucketer());
+				.setBucketer(new KeyBucketer())
+				.setBatchSize(Long.MAX_VALUE)
```

Comment:
What it the reason for these changes? Should the cause of the problem be already solved by having an infinite source now?

Reply:
Actually these are the real instability fix. These make sure that there are no `pending` files between checkpoints which may be invalidated in case of a failure. These files may cause instabilities, given that we _cannot_ safely clean up invalidated `pending` files.